### PR TITLE
Start stop example

### PR DESCRIPTION
**Pull request by jimmy-zendar**
an example for starting, running and stopping the radar pro grammatically

### CODE REVIEW

File: examples/start_stop/main.cc [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+
+void
+SpinHK()
+{
+  while (auto next_hk = ZenApi::NextHousekeepingReport()) {
+    (void)next_hk;
+  }
+}
+
+} ///< \namespace
```
Comment (hamlittle) [inline examples/start_stop/main.cc:98]: Unused?

File: examples/start_stop/main.cc [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+  default_install_options.install_path = FLAGS_install_path;
+  ZenApi::Start(FLAGS_imaging_mode, default_install_options, FLAGS_stream_addr);
+
+  std::cout << "radar started" << std::endl;
+
+  // subscribe to radar data
+  ZenApi::SubscribeImages();
+  ZenApi::SubscribeTrackerStates();
+  ZenApi::SubscribeLogMessages();
+  ZenApi::SubscribeHousekeepingReports();
```
Comment (hamlittle) [inline examples/start_stop/main.cc:143]: If `SpinHK()` is unused this is safe to remove.

File: examples/start_stop/main.cc [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+
+  // listen to data
+  auto image_reader = std::thread(SpinImages, &image_counter);
+  auto points_reader = std::thread(SpinPoints, &pc_counter);
+  auto log_reader = std::thread(SpinLogs, &log_file);
+  auto hk_reader = std::thread(SpinHK);
```
Comment (hamlittle) [inline examples/start_stop/main.cc:149]: Ditto here.

File: examples/start_stop/main.cc [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+
+  std::this_thread::sleep_for(
+      std::chrono::seconds(std::stoi(FLAGS_run_duration)));
+
+  // close down
+  ZenApi::UnsubscribeImages();
+  ZenApi::UnsubscribeTrackerStates();
+  ZenApi::UnsubscribeHousekeepingReports();
```
Comment (hamlittle) [inline examples/start_stop/main.cc:157]: Ditto here.

File: examples/start_stop/main.cc [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+  stream_addr, "localhost",
+  "Where the device should stream the image data. "
+  "**Note** This flag tells the device where to send the image data, so this "
+  "address should be from the device's viewpoint."
+);
+
+DEFINE_string(
+  run_duration, "120",
+  "How long (in seconds) to run this program before disconnect. "
+);
```
Comment (hamlittle) [inline examples/start_stop/main.cc]: Suggest `DEFINE_int32` instead of `DEFINE_string`

File: examples/start_stop/main.cc [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+  ZenApi::SubscribeHousekeepingReports();
+
+  // listen to data
+  auto image_reader = std::thread(SpinImages, &image_counter);
+  auto points_reader = std::thread(SpinPoints, &pc_counter);
+  auto log_reader = std::thread(SpinLogs, &log_file);
+  auto hk_reader = std::thread(SpinHK);
+
+  std::this_thread::sleep_for(
+      std::chrono::seconds(std::stoi(FLAGS_run_duration)));
```
Comment (hamlittle) [inline examples/start_stop/main.cc]: Suggest to change the type from string to int32, and then won't need `std::stoi(...)`. See previous comment.

File: examples/start_stop/main.cc [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+  ZenApi::UnsubscribeTrackerStates();
+  ZenApi::UnsubscribeHousekeepingReports();
+  ZenApi::UnsubscribeLogMessages();
+
+  ZenApi::Release();
+  image_reader.join();
+  points_reader.join();
+
+  ZenApi::Disconnect();
+  hk_reader.join();
```
Comment (hamlittle) [inline examples/start_stop/main.cc:165]: Ditto here.

**Review by hamlittle — APPROVED** [review]
LGTM.

Left a couple comments about some code clean up.

File: examples/start_stop/main.cc (hunk 5 above)
Comment (osamasal) [reply examples/start_stop/main.cc]: Agree.

File: examples/start_stop/main.cc [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+  while (auto next_log = ZenApi::NextLogMessage()) {
+    std::cout << next_log->canonical_message() << std::endl;
+    if (next_log->severity() > 0){
+      (*file) << next_log->canonical_message() << std::endl;
+    }
+  }
+}
+
+void
+SpinHK()
```
Comment (osamasal) [inline examples/start_stop/main.cc:91]: What is this housekeeping report for?

**Review by osamasal — APPROVED** [review]
Looks reasonable, just minor comments.

File: examples/start_stop/main.cc (hunk 5 above)
Comment (jimmy-zendar) [reply examples/start_stop/main.cc]: Thank you for catching it.

**Pull request merged**